aea/protocols/default/serialization.py

```diff
@@ -68,4 +68,4 @@ def decode(self, obj: bytes) -> Message:
         else:
             raise ValueError("Type not recognized.")
 
-        return DefaultMessage(body=body)
+        return DefaultMessage(type=msg_type, body=body)
```

Comment:
do we need to change this on the other protocols too?

Reply:
yes! good point.